### DISCUSSION TITLE
Does LibKTX depend on Java libraries?

### DISCUSSION THREAD

**Discussion by SohnyBohny** [libktx/ktx#116] [Q&A] [answered]
Hey,

I have a question:
Is libkdx pure Kotlin or does it depend one some Java libraries?

I'm asking because of Kotlin's js support.

"When you choose the JavaScript target, any Kotlin code that is part of the project as well as the standard library that ships with Kotlin is transpiled to JavaScript. However, this excludes the JDK and any JVM or Java framework or library used. Any file that is not Kotlin will be ignored during compilation." ([Kotlin docs](https://kotlinlang.org/docs/reference/js-overview.html))

Is it possible to compile a libkdx project written in Kotlin to Javascript in the actual state of the project or anytime in the future?

Thank you :smile:

**Comment by czyzby** [comment, marked as answer]
KTX itself is built around [LibGDX](https://libgdx.badlogicgames.com/), which is written in Java with some C++. You will not be able to use Kotlin-to-JS because of the crucial LibGDX dependency.

However, LibGDX itself works in the browser. The official backend uses GWT, which is only able to compile Java _code_, but there are two open source projects (that I know of) that use tools which can compile JVM _bytecode_, supporting other languages. Look into [Dragome](https://github.com/xpenatan/gdx-dragome-backend) and [TeaVM](https://github.com/konsoletyper/teavm-libgdx).

**Comment by SohnyBohny** [comment]
Okay - thank you :smiley:
